src/sentry/utils/suspect_resolutions/get_suspect_resolutions.py

```diff
@@ -1,13 +1,21 @@
 from typing import Sequence
 
 from sentry.models import Activity, Group, GroupStatus
+from sentry.signals import issue_resolved
+from sentry.tasks.base import instrumented_task
 from sentry.utils.suspect_resolutions import analytics
 from sentry.utils.suspect_resolutions.commit_correlation import is_issue_commit_correlated
 from sentry.utils.suspect_resolutions.metric_correlation import is_issue_error_rate_correlated
 
 
-def get_suspect_resolutions(resolved_issue: Group) -> Sequence[Group]:
+@issue_resolved.connect(weak=False)
+def record_suspect_resolutions(organization_id, project, group, user, resolution_type, **kwargs):
+    get_suspect_resolutions.delay(group.id)
 
+
+@instrumented_task(name="sentry.tasks.get_suspect_resolutions", queue="get_suspect_resolutions")
+def get_suspect_resolutions(resolved_issue_id: int) -> Sequence[Group]:
+    resolved_issue = Group.objects.filter(id=resolved_issue_id).first()
```

Comment:
Nit: `Group.objects.get(id=resolved_issue_id)`

Reply:
Curious, I've been doing the `filter().first()` pattern over `try .. get() .. except DoesNotExist` is there any reason to prefer `get()`?

Reply:
To some extent its personal preference. One different between `.first` and `.get` is that first will add an order by `id` if no order by has been applied to the queryset. In some cases, this can cause your queries to be unexpectedly slow.

There wouldn't be a problem here since we're filtering by id, but I generally prefer to use `.get` when I know I'm fetching a single item.